### PR TITLE
remember password for re-use

### PR DESCRIPTION
**Pull request by reidpr**
This PR remembers user-entered password for the life of the process (addresses #956) and documents the `CH_IMAGE_USERNAME` and `CH_IMAGE_PASSWORD` environment variables (addresses #849).

### CODE REVIEW

**Review by trandles-lanl — APPROVED** [review]
This looks ok to me. The caveats are well documented.

File: doc/ch-image_desc.rst [hunk 1]
```diff
@@ -88,6 +91,34 @@ Common options placed before the sub-command:
...
+:code:`ch-image` checks if something exists, then re-authenticate when
+upgrading the token to read-write for upload. If your site uses one-time
+passwords such as provided by a security device, you can specify
+:code:`--password-once` to provide a new secret each time.
+
+These values are not saved persistently, e.g. in a file. Note that we do use
+normal Python variables for this information, without pinning them into
+physical RAM with `mlock(2)
+<https://man7.org/linux/man-pages/man2/mlock.2.html>`_ or any other special
+treatment, so we cannot guarantee they will never reach non-volatile storage.
```
Comment (heasterday) [inline doc/ch-image_desc.rst:114]: @mej Do you have any thoughts on how risky this is?

**Review by heasterday — COMMENTED** [review]
This looks good to me. I will do some local tests and report back if I bump into anything.

File: doc/ch-image_desc.rst (hunk 1 above)
Comment (reidpr) [reply doc/ch-image_desc.rst:114]: For a little more context, it seemed impossible to reliably `mlock(2)` Python variables. The closest it seemed I could come was to allocate a raw buffer and put the data in there, but then I couldn't get it in or out without going through Python variables anyway.

**Review by heasterday — APPROVED** [review]
Did some basic testing and it looks good

**Pull request merged**